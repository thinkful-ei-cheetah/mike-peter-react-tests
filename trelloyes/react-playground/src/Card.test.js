import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';
import STORE from './store.js';

//this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  //render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Card title={STORE.allCards.a.title} content={STORE.allCards.a.content} />, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Card title={STORE.allCards.a.title} content={STORE.allCards.a.content} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});

it('renders the UI as expected with no unreads', () => {
  const tree = renderer
    .create(<Card title={STORE.allCards.a.title} content={STORE.allCards.a.content} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});