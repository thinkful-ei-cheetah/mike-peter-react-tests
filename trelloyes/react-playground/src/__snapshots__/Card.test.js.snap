// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="card"
>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`renders the UI as expected with no unreads 1`] = `
<div
  className="card"
>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
